typescript/sdk/src/token/checker.ts

```diff
@@ -126,6 +103,29 @@ export class HypERC20Checker extends ProxiedRouterChecker<
         this.addViolation(violation);
       }
     }
+
+    // Check all actual decimals are consistent
+    const actualChainDecimals = await this.getEvmActualDecimals();
```

Comment:
>  This will avoid us calling .decimals on a contract that is not a token contract i.e if the config has been misconfigured

Is this because one of the prior calls will throw or something? As I don't see an early return

Reply:
For a collateral token, we will call .decimal() on it when we go down this code path. If the config in code was setup incorrectly such that the collateral ER20 is not in fact and ERC20, we throw and unhandled error which is not so easy to understand.  

We could instead wrap the decimal call in a try catch and then throw with a better error message but reordering these checks ensures the the collateral ERC20 in the config in code matched on chain state before calling decimals